cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -1166,9 +1166,9 @@ private Future<MapAndStatus<ConfigMap, KafkaRebalanceStatus>> onReady(Reconcilia
                                 new NoSuchResourceException("Kafka resource '" + clusterName
                                         + "' identified by label '" + Labels.STRIMZI_CLUSTER_LABEL
                                         + "' does not exist in namespace " + clusterNamespace + ".")).mapEmpty();
-                    } else if (kafka.getStatus() == null
+                    } else if ((kafka.getStatus() == null
                             || kafka.getStatus().getConditions() == null
-                            || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) {
+                            || kafka.getStatus().getConditions().stream().noneMatch(condition -> condition.getType().equals("Ready") && condition.getStatus().equals("True"))) && state(kafkaRebalance) != KafkaRebalanceState.Ready) {
```

Comment:
@kanha-gupta As @ppatierno said, we can create a separate `isKafkaClusterReady` which can help us get rid of this big if block. I think it would be good to use it for cleaner code.

Reply:
Yes I think that right now the code is unreadable. What I suggested would help.